model/dce-stdio.cc

```diff
@@ -460,6 +412,19 @@ int dce_fileno (FILE *stream)
   NS_LOG_FUNCTION (Current () << UtilsGetNodeId () << stream);
   NS_ASSERT (Current () != 0);
   Thread *current = Current ();
+
+  // UGLY: FIXME: Use current->process->openStreams instead
+  if (current->process->pstdin && stream == *current->process->pstdin) {
+      return 0;
+  } else if (current->process->pstdout && stream == *current->process->pstdout) {
+      return 1;
+  } else if (current->process->pstderr && stream == *current->process->pstderr) {
+      return 2;
+  }
+
+  // FIXME: Handle fopencookie things to. We need to detect those FILE*
+  // and return cookie->_fileno instead... But how?
+
```

Comment:
Can this be fixed by overwriting the ```_fileno``` field of the FILE pointer returned from the ```dce_fdopen()``` function before the final pointer is returned by it. 
Something like ```file->_fileno = cookie->_fileno``` or maybe directly ```file->_fileno = fildes``` ?